### PR TITLE
Feature/screen sharing v2

### PR DESCRIPTION
**Pull request by jordanreedie**
Second pass at screen sharing

## Description
- Change implementation so the user who is sharing their screen can still see the other participants in the meeting. Now, the shared screen replaces the user's local video stream in the top left corner of the app (ie users can no longer see themselves while sharing their screen)
- Inform users if their browser is not capable of screen sharing
- Fix bug where audio would come through local shared screen, resulting in the user being able to hear their own voice when sharing their screen
- Fix bug where if the user refreshed while someone else was sharing their screen, the whole app would break

Note that there is an accompanying change to the SimpleWebRTC repo.

Also, I added the detect-browser node package - is that okay with you @mlippert?

## Motivation and context
Gotta have screen sharing!

## How has this been tested?
Minimally. Tested locally, and everything seems to work all right

## Types of changes
- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ New feature (non-breaking change which adds functionality)

## Checklist:

- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**

### CODE REVIEW

**Comment by jordanreedie** [comment]
Actually, just found a bug: audio does not mute properly when the user is screen sharing

**Comment by jordanreedie** [comment]
ok i fixed the bug with the latest changes to the localmedia library

File: package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "bunyan": "^1.8.12",
     "config": "^3.0.1",
     "cookie-parser": "^1.4.3",
+    "detect-browser": "^4.0.3",
```
Comment (mlippert) [inline package.json]: You asked about adding this package. It's fine, if there's a package that's helpful, go ahead and add it. However, I've been only putting packages used by the server code in dependencies, all other packages that are only used for building the client, I've been putting in the `devDependencies`.

File: src/components/Chat/index.js [hunk 2]
```diff
@@ -137,9 +141,9 @@ const mapDispatchToProps = dispatch => ({
     } else if (webRtcRemoteSharedScreen) {
       // someone is already sharing
       // TODO tell user to quit it
-      console.log("Stop that, someone is already sharing!");
+      logger.debug("Stop that, someone is already sharing!");
```
Comment (mlippert) [inline src/components/Chat/index.js]: thanks, for this one and the others.

File: src/components/Chat/index.js [hunk 3]
```diff
@@ -368,10 +374,116 @@ class Chat extends Component {
...
+  }
+
+  screenShareWarning() {
+    // inform the user screen sharing is not available on their device
+    let text = '';
+    let experimental = false;
+    let version = parseInt(browser.version.split('.')[0])
+    let alertText = 'Copy and paste "chrome://flags/#enable-experimental-web-platform-features" into your '
+    alertText +=    'address bar, click its "Enable" link, and restart Chrome.'
+    switch (browser && browser.name) {
```
[the remark this answers is not in the document]
Comment (jordanreedie) [reply src/components/Chat/index.js]: whoops 😅

File: package.json (hunk 1 above)
Comment (jordanreedie) [reply package.json]: ok great, thanks!

File: src/components/Chat/index.js (hunk 2 above)
Comment (jordanreedie) [reply src/components/Chat/index.js]: for sure!

**Comment by jordanreedie** [comment]
Ok, I think this should be ready to merge if you're satisfied with the changes @mlippert

**Pull request merged**